### PR TITLE
Fix wells for null pointers

### LINKED ISSUE #741: flow crashes for cases without any wells.
Apparently null well pointers are not handeled correctly any more.

To reproduce run the EQUILIBRIUM case from opm-data.

### PR DESCRIPTION
**Pull request by blattms**
There were variuous places in the code that had problems with null well pointers. These are fixed with this PR. Together with OPM/opm-core#1050 this PR closes #741

### CODE REVIEW

File: opm/autodiff/BlackoilModelBase_impl.hpp [hunk 1]
```diff
@@ -223,7 +223,7 @@ namespace detail {
 
         // TODO: put this for now to avoid modify the following code.
         // TODO: this code can be fragile.
-        const Wells* wells_arg = &(asImpl().well_model_.wells());
+        const Wells* wells_arg = asImpl().well_model_.wellsPointer();
```
Comment (atgeirr) [inline opm/autodiff/BlackoilModelBase_impl.hpp]: This variable is not used except in the HAVE_MPI branch, could you move it inside while here? Would fix warnings when not using mpi, and be a little clearer.

File: opm/autodiff/StandardWells_impl.hpp [hunk 2]
```diff
@@ -365,6 +374,7 @@ namespace Opm
             return;
         } else {
             const std::vector<int>& well_cells = wellOps().well_cells;
+            const int num_phases_ = wells().number_of_phases;
```
Comment (atgeirr) [inline opm/autodiff/StandardWells_impl.hpp]: The variable name should not end with an underscore, even when that causes a few more lines below to require the change, too.

**Comment by atgeirr** [comment]
All in all very good, should be merged after fixup.

File: opm/autodiff/StandardWells_impl.hpp (hunk 2 above)
Comment (blattms) [inline opm/autodiff/StandardWells_impl.hpp]: trailing underscore removed.

**Pull request merged**